storage/memory_storage.go

```diff
@@ -32,22 +33,23 @@ func (m *memoryStorageImplementation) Get(key string) (string, error) {
 		return "", fferr.NewInvalidArgumentError(fmt.Errorf("key is empty"))
 	}
 
-	value, ok := m.storage[key]
+	value, ok := m.storage.Load(key)
 	if !ok {
 		return "", fferr.NewKeyNotFoundError(key, nil)
 	}
 
-	return value, nil
+	return value.(string), nil
 }
 
 func (m *memoryStorageImplementation) List(prefix string) (map[string]string, error) {
 	result := make(map[string]string)
 
-	for key, value := range m.storage {
-		if strings.HasPrefix(key, prefix) {
-			result[key] = value
+	m.storage.Range(func(key, value interface{}) bool {
```

Comment:
We need a special lock to do this. Maybe a lock that locks the entire implementation or something.

.Range is not guaranteed to give a consistent result.

Reply:
will look into it

Reply:
So it looks like the locker does handle prefixes